packages/framework-provider-azure-infrastructure/src/infrastructure/helper/function-zip.ts

```diff
@@ -0,0 +1,117 @@
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { GraphqlFunction } from '../functions/graphql-function'
+import { EventHandlerFunction } from '../functions/event-handler-function'
+import { SchedulesFunctions } from '../functions/schedules-functions'
+import { FunctionDefinition } from '../types/functionDefinition'
+import archiver = require('archiver')
+import * as fs from 'fs'
+import * as path from 'path'
+import * as os from 'os'
+import { ZipResource } from '../types/zip-resource'
+import needle = require('needle')
+import { azureCredentials, createWebSiteManagementClient } from './utils'
+import { User } from 'azure-arm-website/lib/models'
+
+export class FunctionZip {
+  static async deployZip(
+    functionAppName: string,
+    resourceGroupName: string,
+    zipResource: ZipResource
+  ): Promise<ZipResource> {
+    const credentials = await FunctionZip.getCredentials(resourceGroupName, functionAppName)
+    await FunctionZip.deployFunctionPackage(
+      zipResource.path,
+      credentials.publishingUserName,
+      credentials.publishingPassword ?? '',
+      credentials.name ?? ''
+    )
+    return zipResource
+  }
+
+  static async copyZip(config: BoosterConfig): Promise<ZipResource> {
+    const zipPath = await this.zipAzureFunction(config)
+    const originFile = path.basename(zipPath)
+    const destinationFile = path.join(process.cwd(), originFile)
+    fs.copyFileSync(zipPath, destinationFile)
+
+    return { name: 'functions', path: destinationFile, fileName: originFile }
+  }
+
+  private static async getCredentials(resourceGroupName: string, functionAppName: string): Promise<User> {
+    const credentials = await azureCredentials()
+    const webSiteManagementClient = await createWebSiteManagementClient(credentials)
+    return await webSiteManagementClient.webApps.listPublishingCredentials(resourceGroupName, functionAppName)
+  }
+
+  private static async deployFunctionPackage(
+    packagePath: string,
+    username: string,
+    password: string,
+    functionAppName: string
+  ): Promise<any> {
+    needle.defaults({
+      open_timeout: 0,
+    })
+
+    return needle(
+      'post',
+      `https://${functionAppName}.scm.azurewebsites.net/api/zipDeploy`,
```

Comment:
Maybe this should be a function that generates the name, so its easier to change if needed?

Reply:
Yep!